tests/components/smartthings/test_switch.py

```diff
@@ -120,3 +124,71 @@ async def test_state_update(
     )
 
     assert hass.states.get("switch.2nd_floor_hallway").state == STATE_OFF
+
+
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+@pytest.mark.parametrize(
+    ("device_fixture", "entity_id"),
+    [
+        ("da_wm_wm_000001", "switch.washer"),
+        ("da_wm_wd_000001", "switch.dryer"),
+    ],
+)
+async def test_create_issue(
+    hass: HomeAssistant,
+    devices: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    issue_registry: ir.IssueRegistry,
+    entity_id: str,
+) -> None:
+    """Test we create an issue when an automation or script is using a deprecated entity."""
+    issue_id = f"deprecated_switch_{entity_id}"
+
+    assert await async_setup_component(
+        hass,
+        automation.DOMAIN,
+        {
+            automation.DOMAIN: {
+                "alias": "test",
+                "trigger": {"platform": "state", "entity_id": entity_id},
+                "action": {
+                    "action": "automation.turn_on",
+                    "target": {
+                        "entity_id": "automation.test",
+                    },
+                },
+            }
+        },
+    )
+    assert await async_setup_component(
+        hass,
+        script.DOMAIN,
+        {
+            script.DOMAIN: {
+                "test": {
+                    "sequence": [
+                        {
+                            "condition": "state",
+                            "entity_id": entity_id,
+                            "state": "on",
+                        },
+                    ],
+                }
+            }
+        },
+    )
+
+    await setup_integration(hass, mock_config_entry)
+
+    assert automations_with_entity(hass, entity_id)[0] == "automation.test"
+    assert scripts_with_entity(hass, entity_id)[0] == "script.test"
+
+    assert len(issue_registry.issues) == 1
```

Comment:
Can we instead verify that the correct issue is created and that the placeholders match the expected entities